processor/k8sattributesprocessor/internal/kube/client.go

```diff
@@ -639,3 +657,63 @@ func (c *WatchClient) extractNamespaceLabelsAnnotations() bool {
 func needContainerAttributes(rules ExtractionRules) bool {
 	return rules.ContainerImageName || rules.ContainerImageTag || rules.ContainerID
 }
+
+func (c *WatchClient) handleReplicasetAdd(obj interface{}) {
+	if replicaset, ok := obj.(*apps_v1.ReplicaSet); ok {
+		c.addOrUpdateReplicaset(replicaset)
+	} else {
+		c.logger.Error("object received was not of type apps_v1.ReplicaSet", zap.Any("received", obj))
+	}
+}
+
+func (c *WatchClient) handleReplicasetUpdate(old, new interface{}) {
+	if replicaset, ok := new.(*apps_v1.ReplicaSet); ok {
+		c.addOrUpdateReplicaset(replicaset)
+	} else {
+		c.logger.Error("object received was not of type apps_v1.ReplicaSet", zap.Any("received", new))
+	}
+}
+
+func (c *WatchClient) handleReplicasetDelete(obj interface{}) {
+	if replicaset, ok := obj.(*apps_v1.ReplicaSet); ok {
+		c.m.Lock()
+		key := string(replicaset.UID)
+		delete(c.Replicasets, key)
+		c.m.Unlock()
+	} else {
+		c.logger.Error("object received was not of type apps_v1.ReplicaSet", zap.Any("received", obj))
+	}
+}
+
+func (c *WatchClient) addOrUpdateReplicaset(replicaset *apps_v1.ReplicaSet) {
+	newReplicaset := &Replicaset{
+		Name:      replicaset.Name,
+		Namespace: replicaset.Namespace,
+		UID:       string(replicaset.UID),
+	}
+
+	if len(replicaset.OwnerReferences) > 0 {
+		deployment := &Deployment{
+			Name: replicaset.OwnerReferences[0].Name,
+			UID:  string(replicaset.OwnerReferences[0].UID),
+		}
+		newReplicaset.Deployment = deployment
+	}
+
+	c.m.Lock()
+	if replicaset.Name != "" {
+		c.Replicasets[string(replicaset.UID)] = newReplicaset
+	}
+	c.m.Unlock()
+}
+
+// GetReplicaset takes replicaset uild
```

Comment:
I suspect `uild` is a typo of `uid`, but if not, could you clear up this comment? If it is a typo, I think the comment can probably be removed.

Reply:
It's typo. =D